src/character_functions.cpp

```diff
@@ -341,8 +340,10 @@ int rate_sleep_spot( const Character &who, const tripoint &p )
 {
     const int current_stim = who.get_stim();
     const comfort_response_t comfort_info = base_comfort_value( who, p );
-    if( comfort_info.aid != nullptr ) {
-        who.add_msg_if_player( m_info, _( "You use your %s for comfort." ), comfort_info.aid->tname() );
+    if( !comfort_info.aid.empty() ) {
+        for( item *comfort_item : comfort_info.aid ) {
+            who.add_msg_if_player( m_info, _( "You use your %s for comfort." ), comfort_item->tname() );
+        }
     }
```

Comment:
this if loop is unnecessary because for loop does nothing on empty array

```suggestion
      for( item *comfort_item : comfort_info.aid ) {
        who.add_msg_if_player( m_info, _( "You use your %s for comfort." ), comfort_item->tname() );
      }
```

Reply:
Ah, makes sense.